### PR TITLE
Migrate `rustc_mir_build` diagnostics

### PR DESCRIPTION
**Pull request by lzcunt**
This PR migrates `rustc_mir_build` to use translatable diagnostics
@rustbot label +A-translation
r? rust-lang/diagnostics

### CODE REVIEW

File: compiler/rustc_mir_build/src/lib.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@ extern crate rustc_middle;
```
Comment (davidtwco) [inline compiler/rustc_mir_build/src/lib.rs:17]: What is remaining for this to deny the migration lints?

**Review by davidtwco — COMMENTED** [review]
This looks good to me, one question..

File: compiler/rustc_mir_build/src/lib.rs (hunk 1 above)
Comment (TheOddGarlic) [reply compiler/rustc_mir_build/src/lib.rs:17]: There's only more unsafe op lints (which I've just migrated rn, those were tricky and took some time bc I had to split a single lint into different lints for every unsafe op), and pattern lints

**Comment by davidtwco** [comment]
Use `@rustbot ready` when you're looking for a review :)

**Review by davidtwco — COMMENTED** [review]
Still looks good, I assume you're still working on it as the PR is a draft, nice work!

**Comment by TheOddGarlic** [comment]
Thank you, and yes I'm still working actively on this. There are more pattern diagnostics than I imagined at first. Also I've been committing most diagnostics separately, should I squash them or keep them this way?

**Comment by davidtwco** [comment]
> Thank you, and yes I'm still working actively on this. There are more pattern diagnostics than I imagined at first. Also I've been committing most diagnostics separately, should I squash them or keep them this way?

Either is fine :)

**Comment by TheOddGarlic** [comment]
I've hit a problem where the most recent subdiagnostic's arguments "shadow" the previous ones (this was previously mentioned in Zulip) while migrating `rustc_mir_build` and it also looks like a lot of the diagnostics not yet migrated in `rustc_mir_build` depend on lists

**Comment by davidtwco** [comment]
> I've hit a problem where the most recent subdiagnostic's arguments "shadow" the previous ones (this was previously mentioned in Zulip) while migrating `rustc_mir_build` and it also looks like a lot of the diagnostics not yet migrated in `rustc_mir_build` depend on lists

We'll just need to leave this untranslated for now, I'm afraid, so feel free to skip them and leave `#[allow(..)]` attributes for the lints (or if there really are a lot, just don't add the `#[deny(..)]`).

**Comment by davidtwco** [comment]
Oh, and apologies for the delay in responding, don't know how I managed to miss this one for six days.

**Comment by TheOddGarlic** [comment]
It happens, I'll quickly resolve everything then

**Comment by Nilstrieb** [comment]
Hi, I've seen you changed some diagnostic structs in your PR. After #103345, the way we refer to fluent messages changed. They are now in a flat namespace with the same identifier as in the fluent file. For example,  `parser::cool_thing` is now `parser_cool_thing` and `parser::suggestion` just `suggestion`.
You should [rebase](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts) to the latest master and change your fluent message references as described above. Thanks!

**Comment by AsyaTheAbove** [comment]
> @AsyaTheAbove are you still able to work on this?

not really, I've been busy recently and eve. when I do get to work on it I have a few issues with my development environment that I've been struggling with

**Comment by mejrs** [comment]
Does this only need a rebase? It seems good to go otherwise.

**Comment by davidtwco** [comment]
Closing in favour of #104417.

**Pull request closed without merging**